searx/engines/google_images.py

```diff
@@ -1,97 +1,205 @@
+        raise RuntimeWarning('sorry.google.com')
+
+    if resp_url.path.startswith('/sorry'):
+        raise RuntimeWarning(gettext('CAPTCHA required'))
+
+    # which subdomain ?
+    # subdomain = resp.search_params.get('google_subdomain')
+
+    # convert the text to dom
     dom = html.fromstring(resp.text)
+    img_bas64_map = scrap_out_thumbs(dom)
 
     # parse results
-    for result in dom.xpath('//div[contains(@class, "rg_meta")]/text()'):
+    #
+    # root element::
+    #     <div id="islmp" ..>
+    # result div per image::
+    #     <div jsmodel="tTXmib"> / <div jsaction="..." data-id="..."
+    #     The data-id matches to a item in a json-data structure in::
+    #         <script nonce="I+vqelcy/01CKiBJi5Z1Ow">AF_initDataCallback({key: 'ds:1', ... data:function(){return [ ...
+    #     In this structure the ling to the origin PNG, JPG or whatever is given
+    #     (we do not blow out the link there, you could still implement that)
+    # first link per image-div contains a <img> with the data-iid for bas64 encoded image data::
+    #      <img class="rg_i Q4LuWd" data-iid="0"
+    # second link per image-div is the target link::
+    #      <a class="VFACy kGQAp" href="https://en.wikipedia.org/wiki/The_Sacrament_of_the_Last_Supper">
+    # the second link also contains two div tags with the *description* and *publisher*::
+    #      <div class="WGvvNb">The Sacrament of the Last Supper ...</div>
+    #      <div class="fxgdke">en.wikipedia.org</div>
+
+    root = eval_xpath(dom, '//div[@id="islmp"]')
+    if not root:
+        logger.error("did not find root element id='islmp'")
+        return results
+
+    root = root[0]
+    for img_node in eval_xpath(root, './/img[contains(@class, "rg_i")]'):
 
         try:
-            metadata = loads(result)
-
-            img_format = metadata.get('ity', '')
-            img_width = metadata.get('ow', '')
-            img_height = metadata.get('oh', '')
-            if img_width and img_height:
-                img_format += " {0}x{1}".format(img_width, img_height)
-
-            source = metadata.get('st', '')
-            source_url = metadata.get('isu', '')
-            if source_url:
-                source += " ({0})".format(source_url)
-
-            results.append({'url': metadata['ru'],
-                            'title': metadata['pt'],
-                            'content': metadata.get('s', ''),
-                            'source': source,
-                            'img_format': img_format,
-                            'thumbnail_src': metadata['tu'],
-                            'img_src': metadata['ou'],
-                            'template': 'images.html'})
-
-        except:
+            img_alt = eval_xpath(img_node, '@alt')[0]
+
+            img_base64_id = eval_xpath(img_node, '@data-iid')
+            if img_base64_id:
+                img_base64_id = img_base64_id[0]
+                thumbnail_src = img_bas64_map[img_base64_id]
```

Comment:
Good point, but I ponder about your condition terms. Why not just 

    if url.startswith('data:image/'):
        return url

Reply:
SVG [allows javascript](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/script), so it is a privacy issue.

```and partial_base64[1].startswith('base64,')``` should be removed to fully support the data [URL protocol](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs). But
* before it was ```data:image/jpeg;base64,```
* it is remove we need to check if it working with morty (I thinking about the URL encoding)
* I don't think a website will use URL encoding to send a gif or a jpg (instead of using base64).

Reply:
> SVG allows javascript, so it is a privacy issue.

Wow, thanks for the hint ... I did not aware! Should we add an issue to have an eye on SVG in general?

> So let's keep it for now.

Yes, lets change when we have an issue with :) .. I have only seen jpg in google-News

> the data URL protocol.

Another _Wow_ from mine .. did you read the "History & Security" chapters from [RFC2397](https://tools.ietf.org/html/rfc2397) .. OMG .. but read by yourself ..

````
...It is being used for object parameters in Java and ActiveX applications.
...
   Sites which use firewall proxies to disallow the retrieval of certain
   media types (such as application script languages or types with known
   security problems) will find it difficult to screen against the
   inclusion of such types using the "data" URL scheme.  However, they
   should be aware of the threat and take whatever precautions are
   considered necessary within their domain.
````

Can't say any words more ..

Reply:
Yes it is an inline document, so it can be whatever.
It can be a way to save a generated document from javascript: https://www.npmjs.com/package/file-saver

In Morty, there is a list of allowed MIME type: 
https://github.com/asciimoo/morty/blob/master/morty.go#L48-L70

So, here a more safe version:
```python

    if url.startswith('data:image/'):
        # 50 is an arbitrary number to get only the beginning of the image. 
        partial_base64 = url[len('data:image/'):50].split(';')
        if len(partial_base64) == 2 \
            and partial_base64[0] in ['gif', 'png', 'jpeg', 'pjpeg', 'webp', 'tiff', 'bmp']\
            and partial_base64[1].startswith('base64,'):
            return url
        else:
            return None
```

Reply:
Looks good to me / would be nice if you like to patch it afterwards / thanks!

Reply:
I mark this as _resolved_, if you think it's not, please reopen the thread / tanks!

Reply:
I wish to not break the master branch. Without this patch it is the case.

Reply:
Sorry,  I am not sure I can follow .. do you mean we should apply your patch from above (see _"So, here a more safe version"_)?

Reply:
Sorry for late reply .. I tested it yesterday .. but I want to make a exercise from .. I will get in touch, thanks for your patience

Reply:
PR #2016 is merged into master